src/Compilers/Core/Portable/Diagnostic/Diagnostic_SimpleDiagnostic.cs

```diff
@@ -130,11 +130,15 @@ public override bool Equals(object obj)
 
             public override int GetHashCode()
             {
-                return Hash.Combine(_descriptor,
-                        Hash.Combine(_messageArgs.GetHashCode(),
-                         Hash.Combine(_location.GetHashCode(),
-                          Hash.Combine(_severity.GetHashCode(), _warningLevel)
-                        )));
+                int hash = 0;
```

Comment:
Nit: Assignment to zero is not necessary since it's going to be overwritten on the next line.